src/c-writer.cc

```diff
@@ -1199,12 +1206,77 @@ void CWriter::WriteGetFuncTypeDecl() {
 }
 
 void CWriter::WriteInitExpr(const ExprList& expr_list) {
-  if (expr_list.empty())
-    return;
+  if (expr_list.empty()) {
+    WABT_UNREACHABLE;
+  }
+
+  std::vector<std::string> mini_stack;
+
+  for (const auto& expr : expr_list) {
+    if (expr.type() == ExprType::Binary) {
+      // Extended const expressions include at least one binary op.
+      // This builds a C expression from the operands.
+      if (mini_stack.size() < 2) {
+        WABT_UNREACHABLE;
+      }
+
+      const auto binexpr = cast<BinaryExpr>(&expr);
+      char op;
+      switch (binexpr->opcode) {
+        case Opcode::I32Add:
+        case Opcode::I64Add:
+        case Opcode::F32Add:
+        case Opcode::F64Add:
+          op = '+';
+          break;
+
+        case Opcode::I32Sub:
+        case Opcode::I64Sub:
+        case Opcode::F32Sub:
+        case Opcode::F64Sub:
+          op = '-';
+          break;
+
+        case Opcode::I32Mul:
+        case Opcode::I64Mul:
+        case Opcode::F32Mul:
+        case Opcode::F64Mul:
+          op = '*';
+          break;
+
+        default:
+          WABT_UNREACHABLE;
+      }
+
+      std::string combination =
+          "((" + std::string(GetCTypeName(binexpr->opcode.GetParamType1())) +
+          ")" + mini_stack.at(mini_stack.size() - 2) + ")" + op + "((" +
+          std::string(GetCTypeName(binexpr->opcode.GetParamType2())) + ")" +
+          mini_stack.at(mini_stack.size() - 1) + ")";
+      mini_stack.resize(mini_stack.size() - 2);
+      mini_stack.push_back(move(combination));
+    } else {
+      // Leaf node (nullary const expression)
+      Stream* existing_stream = stream_;
+      MemoryStream terminal_stream;
+      stream_ = &terminal_stream;
```

Comment:
Neat trick.. I would not have though of temporarily overriding the output `stream_` like that.